### PR TITLE
13.0 add new version

### PR DESCRIPTION
**Pull request by grindtildeath**

### CODE REVIEW

File: 13.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+COPY ./before-migrate-entrypoint.d /before-migrate-entrypoint.d
+COPY ./start-entrypoint.d /start-entrypoint.d
+COPY ./MANIFEST.in /odoo
+
+VOLUME ["/data/odoo", "/var/log/odoo"]
+
+# Expose Odoo services
+EXPOSE 8069 8072
+
+ENV ODOO_VERSION=12.0 \
```
Comment (TDu) [inline 13.0/Dockerfile]:
```suggestion
ENV ODOO_VERSION=13.0 \
```

**Comment by leemannd** [comment]
We should update also -> https://github.com/camptocamp/docker-odoo-project/blob/master/travis/publish.sh#L9

File: travis/publish.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
   docker login --username="$DOCKER_USERNAME" --password="$DOCKER_PASSWORD"
 
   if [ "$TRAVIS_BRANCH" == "master" ]; then
-    if [ "$VERSION" == "12.0" ]; then
+    if [ "$VERSION" >= "12.0" ]; then
```
Comment (leemannd) [inline travis/publish.sh]: Hum, this means we are going to push the latest with `$VERSION IN [12.0,13.0]` I'm not sure this is what we want.

File: 13.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    libssl-dev \
+    libjpeg-dev \
+    libpng-dev \
+    zlib1g-dev \
+    git
+
+# Install some deps, lessc and less-plugin-clean-css, and wkhtmltopdf
+RUN set -x; \
+        sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ stretch-pgdg main" > /etc/apt/sources.list.d/pgdg.list' \
+        && /install/package_odoo_11.0_12.0.sh \
```
Comment (leemannd) [inline 13.0/Dockerfile]: I'm not fond of the nomenclature, but it includes the name of all versions where it is used.
Feel free to ignore my nitpicking comment.

**Comment by grindtildeath** [comment]
@guewen Your review would be much appreciated :pray:

File: 13.0/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,78 @@
+FROM debian:stretch
```
Comment (guewen) [inline 13.0/Dockerfile]: Worth to try with `debian:buster`

File: 13.0/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+# Moved because there was a bug while installing `odoo-autodiscover`. There is
+# an accent in the contributor name
+ENV LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8
+
+# build and dev packages
+ENV BUILD_PACKAGE \
+    build-essential \
+    gcc \
+    python3.5-dev \
```
Comment (guewen) [inline 13.0/Dockerfile]: python3.7-dev I guess in debian buster

File: 13.0/base_requirements.txt [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+Jinja2==2.10.1
+libsass==0.12.3
+lxml==3.7.1 ; sys_platform != 'win32' and python_version < '3.7'
+Mako==1.0.4
+MarkupSafe==0.23
+mock==2.0.0
+num2words==0.5.6
+ofxparse==0.16
+passlib==1.6.5
+Pillow==4.0.0
```
Comment (yvaucher) [inline 13.0/base_requirements.txt]:
```suggestion
Pillow==5.4.0
```

File: 13.0/base_requirements.txt [hunk 7]
```diff
@@ -0,0 +1,86 @@
+# Odoo dependencies
+Babel==2.3.4
+chardet==3.0.4
+decorator==4.0.10
+docutils==0.12
+ebaysdk==2.1.5
+feedparser==5.2.1
+gevent==1.1.2 ; sys_platform != 'win32' and python_version < '3.7'
```
Comment (yvaucher) [inline 13.0/base_requirements.txt]:
```suggestion
gevent==1.3.4 ; sys_platform != 'win32' and python_version >= '3.7'
```
We want the 3.7 lib version.

File: 13.0/base_requirements.txt [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+greenlet==0.4.10 ; python_version < '3.7'
```
Comment (yvaucher) [inline 13.0/base_requirements.txt]:
```suggestion
greenlet==0.4.13 ; python_version >= '3.7'
```

Same as gevent, 3.7 is targeted

File: 13.0/base_requirements.txt [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+html2text==2016.9.19
+Jinja2==2.10.1
+libsass==0.12.3
+lxml==3.7.1 ; sys_platform != 'win32' and python_version < '3.7'
```
Comment (yvaucher) [inline 13.0/base_requirements.txt]:
```suggestion
lxml==4.2.3 ; sys_platform != 'win32' and python_version >= '3.7'
```
Same as previous. Prefer 3.7

**Comment by yvaucher** [comment]
12.0 is broken :-/

<details><summary>:warning: Error wkhtmltox.deb: FAILED</summary>

```
wkhtmltox.deb: FAILED
sha1sum: WARNING: 1 computed checksum did NOT match
Selecting previously unselected package wkhtmltopdf.
(Reading database ... 13632 files and directories currently installed.)
Preparing to unpack wkhtmltox.deb ...
Unpacking wkhtmltopdf (0.12.5-1) ...
dpkg: wkhtmltopdf: dependency problems, but configuring anyway as you requested:
 wkhtmltopdf depends on libgl1; however:
  Package libgl1 is not installed.
 wkhtmltopdf depends on libqt5core5a (>= 5.11.0~rc1); however:
  Package libqt5core5a is not installed.
 wkhtmltopdf depends on libqt5gui5 (>= 5.2.0); however:
  Package libqt5gui5 is not installed.
 wkhtmltopdf depends on libqt5network5 (>= 5.0.2); however:
  Package libqt5network5 is not installed.
 wkhtmltopdf depends on libqt5printsupport5 (>= 5.2.0); however:
  Package libqt5printsupport5 is not installed.
 wkhtmltopdf depends on libqt5svg5 (>= 5.6.0~beta); however:
  Package libqt5svg5 is not installed.
 wkhtmltopdf depends on libqt5webkit5 (>= 5.6.0~rc); however:
  Package libqt5webkit5 is not installed.
 wkhtmltopdf depends on libqt5widgets5 (>= 5.0.2); however:
  Package libqt5widgets5 is not installed.
 wkhtmltopdf depends on libqt5xmlpatterns5 (>= 5.0.2); however:
  Package libqt5xmlpatterns5 is not installed.
Setting up wkhtmltopdf (0.12.5-1) ...
+ /install/dev_package.sh
Hit:1 http://security.debian.org/debian-security stretch/updates InRelease
Ign:2 http://deb.debian.org/debian stretch InRelease
Hit:3 http://deb.debian.org/debian stretch-updates InRelease
Hit:4 http://apt.postgresql.org/pub/repos/apt stretch-pgdg InRelease
Hit:5 http://deb.debian.org/debian stretch Release
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 build-essential : Depends: libc6-dev but it is not going to be installed or
                            libc-dev
                   Depends: g++ (>= 4:6.3) but it is not going to be installed
                   Depends: make
                   Depends: dpkg-dev (>= 1.17.11) but it is not going to be installed
 gcc : Depends: cpp (>= 4:6.3.0-4) but it is not going to be installed
       Depends: gcc-6 (>= 6.3.0-9~) but it is not going to be installed
 git : Depends: libcurl3-gnutls (>= 7.16.2) but it is not going to be installed
       Depends: perl but it is not going to be installed
       Depends: liberror-perl but it is not going to be installed
       Depends: git-man (> 1:2.11.0) but it is not going to be installed
       Depends: git-man (< 1:2.11.0-.) but it is not going to be installed
 libevent-dev : Depends: libevent-2.0-5 (= 2.0.21-stable-3) but it is not going to be installed
                Depends: libevent-core-2.0-5 (= 2.0.21-stable-3) but it is not going to be installed
                Depends: libevent-extra-2.0-5 (= 2.0.21-stable-3) but it is not going to be installed
                Depends: libevent-pthreads-2.0-5 (= 2.0.21-stable-3) but it is not going to be installed
                Depends: libevent-openssl-2.0-5 (= 2.0.21-stable-3) but it is not going to be installed
 libfreetype6-dev : Depends: libc6-dev but it is not going to be installed or
                             libc-dev
 libjpeg-dev : Depends: libjpeg62-turbo-dev (>= 1:1.5.1-2) but it is not going to be installed
 libsasl2-dev : Depends: libc6-dev but it is not going to be installed
 libxml2-dev : Depends: libicu-dev but it is not going to be installed
 python3.5-dev : Depends: libpython3.5-dev (= 3.5.3-1+deb9u1) but it is not going to be installed
                 Depends: libpython3.5 (= 3.5.3-1+deb9u1) but it is not going to be installed
                 Depends: libexpat1-dev but it is not going to be installed
 wkhtmltopdf : Depends: libgl1
               Depends: libqt5core5a (>= 5.11.0~rc1) but it is not going to be installed
               Depends: libqt5gui5 (>= 5.2.0) but it is not going to be installed
               Depends: libqt5network5 (>= 5.0.2) but it is not going to be installed
               Depends: libqt5printsupport5 (>= 5.2.0) but it is not going to be installed
               Depends: libqt5svg5 (>= 5.6.0~beta) but it is not going to be installed
               Depends: libqt5webkit5 (>= 5.6.0~rc) but it is not going to be installed
               Depends: libqt5widgets5 (>= 5.0.2) but it is not going to be installed
               Depends: libqt5xmlpatterns5 (>= 5.0.2) but it is not going to be installed
 zlib1g-dev : Depends: libc6-dev but it is not going to be installed or
                       libc-dev
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
The command '/bin/sh -c set -x;         sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ stretch-pgdg main" > /etc/apt/sources.list.d/pgdg.list'         && /install/package_odoo_11.0_12.0.sh         && /install/setup-pip.sh         && /install/postgres.sh         && /install/wkhtml_12_5.sh         && /install/dev_package.sh         && python3 -m pip install --force-reinstall pip setuptools         && pip3 install -r /odoo/base_requirements.txt --ignore-installed         && /install/purge_dev_package_and_cache.sh' returned a non-zero code: 100
+on_exit
+echo 'Cleaning up temporary directory...'
Cleaning up temporary directory...
+rm -rf /tmp/tmp.xXrPxJZqKa
+rm -f /tmp/odoo.tar.gz
Makefile:16: recipe for target 'build' failed
make: *** [build] Error 100
The command "make VERSION=$VERSION build" exited with 2.
```
</details>

**Comment by yvaucher** [comment]
I'll try to relaunch the test. It seems the checksum didn't match. If it doesn't fail again the file was corrupted.

**Comment by grindtildeath** [comment]
@yvaucher Just reverted the change I did in this file since it isn't used for 13.0 anymore.

**Comment by yvaucher** [comment]
Successful build was there: https://travis-ci.org/camptocamp/docker-odoo-project/builds/597552074

Rebasing squashing and merging

**Pull request merged**